src/components/menu/menu-item/menu-item.component.js

```diff
@@ -52,17 +52,33 @@ const MenuItem = ({
 
       return [...childArray.map((innerChild) => innerChild)];
     }
+    if (child?.type === Search) {
+      isChildSearch.current = true;
+    }
 
     return child;
   });
 
+  const focusRef = isChildSearch.current ? childRef : ref;
   useEffect(() => {
     if (focusFromSubmenu === undefined && focusFromMenu) {
-      ref.current.focus();
-    } else if (focusFromSubmenu) {
-      ref.current.focus();
+      focusRef.current.focus();
+    } else if (!submenuContext.blockDoubleFocus && focusFromSubmenu) {
```

Comment:
This can just go being to being `else if (focusFromSubmenu)` now that you've made the change on line 291 in submenu.component.js